app/controllers/smart_proxies_controller.rb

```diff
@@ -108,4 +108,11 @@ def action_permission
         super
     end
   end
+
+  def proxy_resource_base
```

Comment:
How about something along the lines of overriding `resource_base` and using super, something like:
```
base = super
base = base.eager_load(:locations) if SETTINGS[:locations_enabled]
base = base.eager_load(:organizations) if SETTINGS[:organizations_enabled]
base
```